nbgrader/converters/autograde.py

```diff
@@ -188,6 +188,7 @@ def convert_single_notebook(self, notebook_filename: str) -> None:
         self._sanitizing = False
         self._init_preprocessors()
         try:
-            super(Autograde, self).convert_single_notebook(notebook_filename)
+            with utils.setenv(NBGRADER_VALIDATING='1'):
```

Comment:
```suggestion
            with utils.setenv(NBGRADER_VALIDATING='autograde'):
```
What if this was set to a different value, so that one could distinguish the validation and autograde phases if one wanted?

Reply:
Ah yes good idea! I did this slightly differently, by introducing a new environment variable `NBGRADER_EXECUTION` that gives the execution type (either validate or autograde). Then `NBGRADER_VALIDATING` is unchanged, and is only set during validation.